isisd/isis_pdu.c

```diff
@@ -1459,8 +1459,10 @@ int isis_receive(struct thread *thread)
 
 	retval = circuit->rx(circuit, ssnpa);
 
-	if (retval == ISIS_OK)
+#if ISIS_METHOD == ISIS_METHOD_DLPI
```

Comment:
I think the correct check here would be:
`#if ISIS_METHOD != ISIS_METHOD_BPF`

The PF_PACKET method (even though using BPF) still relies on calling `isis_handle_pdu()`, because differently from the `isis_bpf.c` implementation (the one that uses the special device file) it only receives a packet per time. Meanwhile the `isis_bpf.c` implementation can receive multiple packets at the same time.